client/thin-replica-client/include/client/thin-replica-client/grpc_connection.hpp

```diff
@@ -209,6 +212,7 @@ class GrpcConnection {
 
   // gRPC connection
   std::shared_ptr<grpc::Channel> channel_;
+  std::shared_mutex channel_protector_;
```

Comment:
Nit: It might be more explicit if we name it `channel_mutex`.

Reply:
Thanks, @dartdart26, I will update the name of this variable.